src/browser/new-browser.ts

```diff
@@ -108,10 +118,26 @@ export class NewBrowser extends Browser {
         }
     }
 
-    protected _getSessionOpts(): RemoteOptions {
+    protected _isLocalGridUrl(): boolean {
+        return this._config.gridUrl === LOCAL_GRID_URL || getInstance().local;
+    }
+
+    protected async _getSessionOpts(): Promise<RemoteOptions> {
         const config = this._config;
-        const gridUri = new URI(config.gridUrl);
-        const capabilities = this._extendCapabilities(config);
+
+        let gridUrl;
+
+        if (this._isLocalGridUrl() && config.automationProtocol === "webdriver") {
+            gridUrl = await this._getLocalWebdriverGridUrl();
+        } else {
+            // if automationProtocol is not "webdriver", fallback to default grid url from "local"
+            // because in "devtools" protocol we dont need gridUrl, but it still has to be valid URL
+            gridUrl = config.gridUrl === LOCAL_GRID_URL ? DEFAULT_GRID_URL : config.gridUrl;
+        }
```

Comment:
"isLocalGridUrl" === either "gridUrl" is "local", either `testplane` is launched with `--local`
- If isLocalGridUrl == true and automationProtocol === "webdriver", we get webdriver process and use its gridUrl
- if isLocalGridUrl == true and automationProtocol is "devtools", gridUrl won't be used either way, but we need to pass any valid url